packages/ui/src/components/input/select/index.tsx

```diff
@@ -121,10 +120,13 @@ function Component<T extends SelectOption<ValueType>>(
         setOpen(!open);
     }, [open]);
 
-    const getPickHandler = (option: T) => () => {
-        if (onChange) onChange(option);
-        setOpen(false);
-    };
+    const getPickHandler = useCallback(
+        (option: T) => () => {
+            if (onChange) onChange(option);
```

Comment:
Do we need to check if the `onChain` function exists? It's a required prop so maybe it can be removed.

Reply:
You're right, addressed in 4f81b738811921bd08c9548bbeb3f139757e949d